pyworker/worker.py

```diff
@@ -127,6 +126,13 @@ def run(self):
                         self.logger.info('Job %d finished in %d seconds' % \
                             (job.job_id, time_diff))
 
+                # Sleep for a while between each job and break if received SIGTERM
+                try:
+                    time.sleep(self.sleep_delay)
+                except Exception as exception:
+                    if type(exception) == TerminatedException:
+                        break
```

Comment:
Either add `else` and reraise error, or better handle `TerminatedException` only in line 132.